src/redux/missions/missionSlice.js

```diff
@@ -0,0 +1,64 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  missions: [],
+  isLoading: false,
+  error: null,
+};
+
+const client = axios.create({
+  baseURL: 'https://api.spacexdata.com/v3/',
+});
+
+export const fetchMission = createAsyncThunk(
+  'missions/fetchMissions',
+  async (_, thunkAPI) => {
+    try {
+      const response = await client.get('missions');
+      return response.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue({ message: 'Something went wrong' });
+    }
+  },
+);
+
+const missionSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    missionStatus: (state, { payload: { missionId } }) => {
+      const newMissions = state.missions.map((mission) => {
+        if (mission.mission_id === missionId) {
+          return { ...mission, member: !mission.member };
+        }
+        return mission;
+      });
+      return { ...state, missions: newMissions };
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchMission.fulfilled, (state, action) => {
+        const missions = action.payload.map((mission) => ({ ...mission, member: false }));
+        return {
+          ...state,
+          isLoading: false,
+          missions,
+        };
```

Comment:
- [ ] It's perfect that you load your data and trigger your action. But we need to filter the data before using it inside our Redux store.  Please filter the following data and returnthe filter's state. This keeps our data clean and relative.
* mission_id
* mission_name
* description

Reply:
Hello @TadesseDev,
Appreciate your feedback, objects were successfully DE structured, and hence request resolved.

Reply:
Good job @ZewdieMc and @shafiuyushawu, I have to say, your project is really amusing and almost got all the checkpoints of the project requirement. Good job 👌